app/helpers/form_helper.rb

```diff
@@ -7,20 +9,46 @@ def app_setting_value_field_for(app_setting, f)
     end
   end
 
-  def block_text_field_tag(name, value)
-    text_field_tag(name, value, class: 'block col-12 mb2 field')
+  def block_text_field_tag(name, value, options = {})
+    text_field_tag(name, value, options.merge(class: 'block col-12 mb2 field')) +
+      form_input_error_messages(name, options)
   end
 
-  def block_date_field_tag(name, value)
-    date_field_tag(name, value, class: 'block col-12 mb2 field')
+  def block_date_field_tag(name, value, options = {})
+    date_field_tag(name, value, options.merge(class: 'block col-12 mb2 field')) +
+      form_input_error_messages(name, options)
   end
 
-  def us_states_territories_select_tag
+  # rubocop:disable MethodLength
+  def form_input_error_messages(name, options = {})
+    content_tag(:div, nil, class: 'bold red mb2', data: { 'errors-for' => name }) do
+      if options[:required]
+        concat content_tag(
+          :div,
+          t('forms.value_missing'),
+          style: 'display: none',
+          data: { 'errors-when' => 'valueMissing' }
+        )
+      end
+      if options[:pattern]
+        concat content_tag(
+          :div,
+          options[:'data-custom-message'] || t('forms.pattern_mismatch'),
+          style: 'display: none',
+          data: { 'errors-when' => 'patternMismatch' }
+        )
+      end
+    end
+  end
+  # rubocop:enable MethodLength
+
+  def us_states_territories_select_tag(options = {})
     select_tag(
       'state',
       options_for_select(us_states_territories),
-      class: 'block col-12 mb2 field'
-    )
+      options.merge(class: 'block col-12 mb2 field')
+    ) +
+      form_input_error_messages('state', options)
   end
```

Comment:
Since we're already using Simple Form in the app, I would take advantage of it and create [custom inputs](https://github.com/plataformatec/simple_form#custom-inputs) that override the default ones such that they add the `block col-12 mb2 field` class and the error div. Adding classes is as easy as doing this for a specific input:
```ruby
# app/inputs/string_input.rb
class StringInput < SimpleForm::Inputs::StringInput
  def input_html_classes
    super.push('block col-12 mb2 field')
  end
end
```
I believe there's also a way to add a class to all inputs if that's what we want via the [wrappers config](https://github.com/plataformatec/simple_form#configuration).

If we want to disable browser validations across the board, we can add `config.browser_validations = false` in our [Simple Form config](https://github.com/18F/identity-idp/blob/master/config/initializers/simple_form.rb), so we wouldn't need to manually specify `novalidate: true` each time.

The advantage of using Simple Form is that we only have to change the Simple Form inputs, without needing to change the markup of the forms (well, at least not the ones that are already using Simple Form). Developers can keep using the standard form inputs and the correct styling will be automatically applied.

This will also keep our code free of code smells 😄